### DISCUSSION TITLE
Multiple lines selection in transaction view of the store commerce app D365.

### DISCUSSION THREAD

**Discussion by Maison24** [microsoft/Dynamics365Commerce.Solutions#322] [Q&A] [answered]
Hello everyone, I'm trying to implement multiple lines selection on transaction view of the store commerce i have implemented the controller which stacking the lines successfully but its not highlighting the lines in the grid.
The code which I'm sharing same was working perfectly fine in MPOS but in store commerce its not highlighting the cart view grid lines.

Here is my code for the controller.

`import { ProxyEntities } from "PosApi/Entities";
import { IExtensionCartViewControllerContext } from "PosApi/Extend/Views/CartView";
import * as CartView from "PosApi/Extend/Views/CartView";
import { ArrayExtensions, StringExtensions } from "PosApi/TypeExtensions";
export default class CartViewController extends CartView.CartExtensionViewControllerBase {
public static selectedCartLineId: string = StringExtensions.EMPTY;
public static selectedCartLineIds: string[] = [];
public static selectedCartLines: ProxyEntities.CartLine[] = [];
private _selectedCartLines: ProxyEntities.CartLine[];
public _isProcessingAddItemOrCustomer: boolean;

constructor(context: IExtensionCartViewControllerContext) {
    super(context);
    this.cartLineSelectedHandler = (data: CartView.CartLineSelectedData): void => {
        if (!this._selectedCartLines) {
            this._selectedCartLines = [];
        }
      
        // Update selected cart lines
        this._selectedCartLines = data.cartLines;
        // Add highlight to newly selected lines
        this._selectedCartLines.forEach(cartLine => {
            const element = document.querySelector(`[data-line-id="${cartLine.LineId}"]`);
            if (element) {
                element.classList.add('win-selected');
            }
            if (CartViewController.selectedCartLineIds.indexOf(cartLine.LineId) === -1) {
                CartViewController.selectedCartLineIds.push(cartLine.LineId);
                CartViewController.selectedCartLines.push(cartLine);
            }
        });
        if (ArrayExtensions.hasElements(this._selectedCartLines)) {
            this.context.logger.logError("Constructor ->Length: " + this._selectedCartLines.length);
        }
    };
    this.cartLineSelectionClearedHandler = (): void => {
        this.context.logger.logError("Cleared ->Length: " + (this._selectedCartLines ? this._selectedCartLines.length : 0));
        // Clear all selected lines
        this._selectedCartLines.forEach(cartLine => {
            const element = document.querySelector(`[data-line-id="${cartLine.LineId}"]`);
            if (element) {
                element.classList.remove('win-selected');
            }
        });
        this._selectedCartLines = [];
        CartViewController.selectedCartLineIds = [];
        CartViewController.selectedCartLines = [];
    };
}
}`

**Comment by madyke** [comment, marked as answer]
Hello @Maison24 - unfortunately, selecting multiple cart lines in the transaction view is not a supported extension scenario, because the Retail Operation APIs are not designed to operate on a multi-line selection. For example, there would be unexpected behavior trying to void a line or apply a discount if you were to try passing in multiple cart lines.

**Reply by Maison24** [reply]
Hello @madyke 
First of all thank you so much for yours reply. Basically we have customization Related Cash Memo with associated sale representative and customer were using this functionality(Multi Selection) previously in MPOS. But now we have upgraded there system from MPOS to Store Commerce with the same customization code but unfortunately it not reflecting the changes visually.

But it stacking the multiple line detail in a list but not highlighting.

**Reply by madyke** [reply]
Hi @Maison24 - when I discussed this question with our UI expert yesterday, he pointed out that modifying the UI like this and using things such as query selectors was not a supported scenario. He said that's most likely why the customization broke in the transition from MPOS to Store Commerce - since the scenario wasn't supported, we can't guarantee backwards compatibility.

Since multi-line selection isn't supported, there isn't any guidance that the product team can provide here. Any work or customizations done by a partner/customer that aren't supported are changes they make at their own risk, and they are ultimately responsible for ensuring that unsupported changes are updated from version to version and don't break functionality, since the product team isn't aware of them and can't test them during our release process.

**Reply by Maison24** [reply]
Please suggest an alternative solution to implement multi-selection. We are open to using checkboxes at the line level in a custom column or any other approach that can be provided to the customer. The customer's business process relies on multi-selection, and with over 100 stores, we need to offer a workaround to address this issue. Any suggestions would be greatly appreciated.
